modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/webapp/META-INF/resources.xml

```diff
@@ -50,14 +50,14 @@
     <Resource>
         <AuthType>Any</AuthType>
         <HttpVerb>GET</HttpVerb>
-        <Uri>http://localhost:9763/virtual_firealarm/manager/device</Uri>
-        <UriTemplate>/manager/device/*</UriTemplate>
+        <Uri>http://localhost:9763/virtual_firealarm/manager/device/{device_id}</Uri>
+        <UriTemplate>/manager/device/{device_id}</UriTemplate>
     </Resource>
     <Resource>
         <AuthType>Any</AuthType>
         <HttpVerb>GET</HttpVerb>
-        <Uri>http://localhost:9763/virtual_firealarm/manager/devices</Uri>
-        <UriTemplate>/manager/devices/*</UriTemplate>
+        <Uri>http://localhost:9763/virtual_firealarm/manager/devices/{username}</Uri>
```

Comment:
This was added to publish api details into api-publisher when the web-app started. Hence relative-path does not work for api-publisher. This was later replaced to pick up api-details with swagger API annotations.